pkg/storaged/content-views.jsx

```diff
@@ -981,12 +1142,33 @@
                                    }
                                }),
                 ],
+                update: (dlg, vals, trigger) => {
+                    if (vals.purpose == 'block' && (trigger == "layout" || trigger == "pvs" || trigger == "purpose")) {
+                        const max = max_size(vals);
+                        const old_max = dlg.get_options("size").max;
+                        if (vals.size > max || vals.size == old_max)
+                            dlg.set_values({ size: max });
+                        dlg.set_options("size", { max });
+                        dlg.set_options("layout", { explanation: layout_descriptions[vals.layout] });
+                    } else if (trigger == "purpose") {
+                        dlg.set_options("size", { max: vgroup.FreeSize });
+                    }
+                },
                 Action: {
                     Title: _("Create"),
                     action: (vals, progress) => {
-                        if (vals.purpose == "block")
-                            return vgroup.CreatePlainVolume(vals.name, vals.size, { });
-                        else if (vals.purpose == "pool")
+                        if (vals.purpose == "block") {
+                            if (!can_do_layouts || vals.layout == "linear")
+                                return vgroup.CreatePlainVolume(vals.name, vals.size, { });
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17226-20230531-081909-df2e1047-fedora-testing-daily//Coverage/lcov/github-pr.diff.gcov.html)